src/analog.cpp

```diff
@@ -0,0 +1,61 @@
+#include "analog.h"
+
+// assign analog inputs to the correct pins
+AnalogIn acceleratorPedalPin(A6);
+AnalogIn acceleratorSignalPin(A3);
+AnalogIn brakeStatusPin(A1);
+AnalogIn regenerativeBrakingPin(A4);
+
+// initialize the analog inputs' readings
+volatile float acceleratorPedalReading;
+volatile float acceleratorSignalReading;
+volatile float brakeStatusReading;
+volatile float regenerativeBrakingReading;
+
+// read accelerator pedal input
+void readAcceleratorPedal(){
+    acceleratorPedalPin.set_reference_voltage(refrence_voltage);
+    acceleratorPedalReading = acceleratorPedalPin.read_voltage();
+}
+
+// read accelerator signal input
+void readAcceleratorSignal(){
+    acceleratorSignalPin.set_reference_voltage(refrence_voltage);
+    acceleratorSignalReading = acceleratorSignalPin.read_voltage();
+}
+
+// read brake status input
+void readBrakeStatus(){
+    brakeStatusPin.set_reference_voltage(refrence_voltage);
+    brakeStatusReading = brakeStatusPin.read_voltage();
+}
+
+// read regeative braking input
+void readRegeativeBraking(){
+    regenerativeBrakingPin.set_reference_voltage(refrence_voltage);
+    regenerativeBrakingReading = regenerativeBrakingPin.read_voltage();
+}
+
+// read all analog input
+void readAnalog(){
+    readAcceleratorPedal();
+    readAcceleratorSignal();
+    readBrakeStatus();
+    readRegeativeBraking();
+}
+
+// initialize Ticker to setup a recurring interrupt to repeatedly call the function at a specified rate
+Ticker readAnalogDelay;
+
+// 
+void flip() {
+  readAnalog();
+}
+
+// repeatedly call the function with the given parameter
+int initAnalog(std::chrono::microseconds readSignalPeriod) {
```

Comment:
Unless you plan on having meaningful error returns, make this function return void